modules/react/src/utils/use-widget.ts

```diff
@@ -0,0 +1,23 @@
+import {useContext, useMemo, useEffect} from 'react';
+import {DeckGlContext} from './deckgl-context';
+import type {Widget} from '@deck.gl/core';
+
+function useWidget<T extends Widget, PropsT extends {}>(
+  WidgetClass: {new (props: PropsT): T},
+  props: PropsT
+): T {
+  const context = useContext(DeckGlContext);
+  const {widgets, deck} = context;
+  const widget = useMemo(() => new WidgetClass(props), [WidgetClass]);
+
+  widgets?.push(widget);
+  widget.setProps(props);
+
+  useEffect(() => {
+    deck?.setProps({widgets});
```

Comment:
Implication: The widgets prop would be removed from the react API, since this is overriding any user-set widgets.

Reply:
Are you worried about the situation where widgets are specified both via JS and via React? I don't think this should be supported, it seems unnecessarily complex. What happens in a similar case with layers?

Perhaps you could `log.warn` if `useWidget` is called when deck already has widgets defined?

Reply:
I was looking at what [`layers` and `views` ](https://github.com/visgl/deck.gl/blob/e9eada678c485520e0e63cff9436230ae7d34fa8/modules/react/src/utils/extract-jsx-layers.ts#L118)do, and they are merge together vanilla and JSX. This seems trickier to do down in this react component.

Reply:
Fixed by ec8d18fec03407cfeb071becf51b6c1ab2e194ab